src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -114,23 +114,44 @@ public void setProjectId(String projectId) {
   }
 
   public String getZone() {
-    return this.zone;
+    initializeCluster();
+    return this.cluster.getLocation();
   }
 
   @DataBoundSetter
   public void setZone(String zone) {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(zone));
     this.zone = zone;
+    initializeCluster();
   }
 
   public String getClusterName() {
-    return this.clusterName;
+    initializeCluster();
+    return this.cluster.getName();
   }
 
   @DataBoundSetter
   public void setClusterName(String clusterName) {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(clusterName));
     this.clusterName = clusterName;
+    initializeCluster();
+  }
+
+  public String getCluster() {
+    initializeCluster();
+    return ContainerClient.toNameAndLocation(this.cluster);
+  }
+
+  @DataBoundSetter
+  public void setCluster(String cluster) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(cluster));
+    this.cluster = ContainerClient.fromNameAndLocation(cluster);
+  }
+
+  private void initializeCluster() {
+    if (this.cluster == null) {
+      this.cluster = new Cluster().setLocation(this.zone).setName(this.clusterName);
```

Comment:
Overloading the use of the API's Cluster model like this as a struct is discouraged, as it adds a hard dependency on the API model's class structure and this class's implementation details. This approach is very brittle, as any change in the model would break this class. Suggest instead just use logic which maps the ClusterNameLocationString to/from clusterName, zone.